### DISCUSSION TITLE
'magic_enum/magic_enum.hpp' file not found

### DISCUSSION THREAD

**Discussion by janhoogerbrugge** [CHERIoT-Platform/cheriot-rtos#86] [Q&A] [answered]
Hi,

Is there something wrong or am I doing something wrong?

Jan.

cheriot@codespaces-1a3a28:/workspaces/cheriot-rtos/examples/02.hello_compartment$ xmake clean
loading board description from  /workspaces/cheriot-rtos/sdk/boards/sail.json
targeting a simulated environment
using stack high water mark
cheriot@codespaces-1a3a28:/workspaces/cheriot-rtos/examples/02.hello_compartment$ xmake config --sdk=/cheriot-tools/
loading board description from  /workspaces/cheriot-rtos/sdk/boards/sail.json
targeting a simulated environment
using stack high water mark
generating /workspaces/cheriot-rtos/sdk/firmware.ldscript.in ... cache
cheriot@codespaces-1a3a28:/workspaces/cheriot-rtos/examples/02.hello_compartment$ xmake
loading board description from  /workspaces/cheriot-rtos/sdk/boards/sail.json
targeting a simulated environment
using stack high water mark
[ 23%]: cache compiling.release ../../sdk/lib/freestanding/memcmp.c
[ 23%]: cache compiling.release ../../sdk/lib/freestanding/memcpy.c
[ 23%]: cache compiling.release ../../sdk/lib/freestanding/memset.c
[ 25%]: cache compiling.release ../../sdk/core/software_revoker/revoker.cc
[ 27%]: cache compiling.release ../../sdk/core/scheduler/main.cc
[ 29%]: cache compiling.release ../../sdk/lib/atomic/atomic1.cc
error: /workspaces/cheriot-rtos/sdk/include/cheri.hh:11:10: fatal error: 'magic_enum/magic_enum.hpp' file not found
#include <magic_enum/magic_enum.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
  > in ../../sdk/lib/atomic/atomic1.cc
cheriot@codespaces-1a3a28:/workspaces/cheriot-rtos/examples/02.hello_compartment$

**Comment by rmn30** [comment, marked as answer]
Hi, `magic_enum` is included as a submodule (pointer to another repo) so you need to tell git to check it out: `git submodule init && git submodule update` .

**Reply by janhoogerbrugge** [reply]
Thanks!

Now it compiles fine.

**Reply by rmn30** [reply]
Next time you clone you can add `--recurse-submodules` to do this at the same time as cloning. I don't think any of the submodules we use have submodules themselves but, just in, you can combine the init and update steps with `git submodule update --init --recursive`, which will also handle that case.

**Reply by janhoogerbrugge** [reply]
Note that I am using GitHub Code Spaces which did the clone for me.

**Reply by rmn30** [reply]
Yeah, I wonder if we can tell it to clone submodules automatically. Please let us know how you get on!

**Reply by davidchisnall** [reply]
The last time I did it, code spaces did it for me.  If it doesn't, then we should add that to the dev container file.  @janhooferbrugge, would you be able to add that to the [dev container config file](https://github.com/microsoft/cheriot-rtos/blob/main/.devcontainer/devcontainer.json) and raise a PR?

**Reply by davidchisnall** [reply]
Hmm, it looks like we were doing that already.  @rmn30 is going to try to debug it.  Thanks for the report.

**Reply by rmn30** [reply]
Hopefully it will work better once this is merged: https://github.com/microsoft/cheriot-rtos/pull/87

**Reply by davidchisnall** [reply]
Merged.  @janhoogerbrugge, thanks for the report, please can you let us know if it doesn't work?
